### DISCUSSION TITLE
Introducing a feature toggle for JTI in tokenAuth

### DISCUSSION THREAD

**Discussion by alvinangjh** [panva/node-oidc-provider#1262] [Ideas]
In **oidc-provider/lib/shared/token_jwt_auth.js**, the function `getTokenJwtAuth` checks for the presence of `payload.jti`, which is used for Replay Detection. However, [RFC-7519](https://datatracker.ietf.org/doc/html/rfc7519#page-10) states that `jti` is optional. The same statement iss mentioned in several other auth documentation such as [Okta](https://developer.okta.com/docs/api/openapi/okta-oauth/guides/client-auth/). 

Hence, I would like to know if there is any future plan for `jti` to be made optional?

**Comment by panva** [comment]
Client authentication methods `client_secret_jwt` and `private_key_jwt` are defined in [OpenID Connect Core 1.0](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication), the `jti` claim is REQUIRED in these assertions.

> Hence, I would like to know if there is any future plan for jti to be made optional?

There is not. OAuth profiles such as FAPI use the OIDC definition, with `jti` REQUIRED.
